numba/core/cpu.py

```diff
@@ -235,6 +244,8 @@ def get_executable(self, library, fndesc, env):
         - env
             an execution environment (from _dynfunc)
         """
+        if not fndesc.llvm_cpython_wrapper_name:
+            return None
```

Comment:
@apmasell, This is the source of the imputils `AssertionError`. The return value of this function is later used as a key (`func`) in https://github.com/numba/numba/blob/22c322acd943901769ba480fc6d2686642719ae9/numba/core/base.py#L390
So this value needs to be a unique object that is suitable as dict key and will be stored in the Dispatcher as the "entry_point" (`cres.entry_point`) for the overload.

Reply:
The following patch will make it work for `NUMBA_JIT_BACKEND=orc python runtests.py numba.tests.test_builtins.TestOperatorMixedTypes.test_eq_ne` but I am still seeing a different segfault with `NUMBA_JIT_BACKEND=orc python runtests.py numba.tests.test_builtins.TestBuiltins.test_abs`

Btw, the docstring on the function is outdated and misleading.

```diff
diff --git a/numba/core/cpu.py b/numba/core/cpu.py
index 234eb05ffd..7c4e259743 100644
--- a/numba/core/cpu.py
+++ b/numba/core/cpu.py
@@ -235,20 +235,19 @@ class CPUContext(BaseContext):
         """
         Returns
         -------
-        (cfunc, fnptr)
+        cfunc
 
         - cfunc
-            callable function (Can be None)
-        - fnptr
-            callable function address
-        - env
-            an execution environment (from _dynfunc)
+            callable function.
+            It's used as dictionary key thus must uniquely identifiable.
         """
-        if not fndesc.llvm_cpython_wrapper_name:
-            return None
-        # Code generation
-        fnptr = library.get_pointer_to_function(
-            fndesc.llvm_cpython_wrapper_name)
+
+        if fndesc.llvm_cpython_wrapper_name:
+            # Code generation
+            fnptr = library.get_pointer_to_function(
+                fndesc.llvm_cpython_wrapper_name)
+        else:
+            fnptr = 0
 
         # Note: we avoid reusing the original docstring to avoid encoding
         # issues on Python 2, see issue #1908
@@ -259,7 +258,8 @@ class CPUContext(BaseContext):
                                        # objects to keepalive with the function
                                        (library,)
                                        )
-        library.set_env(self.get_env_name(fndesc), env)
+        if fndesc.llvm_cpython_wrapper_name:
+            library.set_env(self.get_env_name(fndesc), env)
         return cfunc
 
     def calc_array_sizeof(self, ndim):
```